source/common/access_log/access_log_manager_impl.cc

```diff
@@ -127,6 +131,10 @@ void AccessLogFileImpl::doWrite(Buffer::Instance& buffer) {
 
 void AccessLogFileImpl::flushThreadFunc() {
 
+  // Local cache of `reopen_file_` so that `reopen_file_` is only used while protected by a mutex,
+  // but the actual reopen operation does not need to happen inside the mutex.
+  bool reopen = false;
```

Comment:
Maybe don't name this "reopen" since there's already a member function named "reopen" in this scope. (Also it's not really clear what it is, and the comment is confusing at best since it says it's a local cache of reopen_file_ but when you assign a value to it below it's the opposite of the value assigned, at the same time, to reopen_file_, which doesn't really resemble the behavior of a cache. I think it's more like you're moving the value from the mutex-guarded one that signals intent, to a local one that signals you're going to do it once the mutex is released. If I'm understanding it right, naming it "reopen_when_unlocked" would make it clearer.)